python/pyiceberg/types.py

```diff
@@ -314,6 +316,10 @@ def __getnewargs__(self) -> Tuple[NestedField, ...]:
         """A magic function for pickling the StructType class."""
         return self.fields
 
+    def __hash__(self) -> int:
```

Comment:
Is there a way to automatically memoize this instead of using a field?

Reply:
Not that I've determined, as python presumes the hash value can change as the instance is mutated.